test/test_array.py

```diff
@@ -2180,6 +2180,39 @@ def test_dtype_conversions(ctx_factory):
 # }}}
 
 
+# {{{ test_svm_mem_pool_with_arrays
+
+def test_svm_mem_pool_with_arrays(ctx_factory):
+    context = ctx_factory()
+    queue = cl.CommandQueue(context)
+    queue2 = cl.CommandQueue(context)
+
+    from pyopencl.characterize import has_coarse_grain_buffer_svm
+    has_cg_svm = has_coarse_grain_buffer_svm(queue.device)
+
+    if not has_cg_svm:
+        pytest.skip("Need coarse-grained SVM support for this test.")
+
+    mem_pool = cl_tools.SVMPool(cl_tools.SVMAllocator(context, alignment=0,
+                                                      queue=queue))
+
+    a_dev = cl_array.arange(queue, 2000, dtype=np.float32, allocator=mem_pool)
+    b_dev = cl_array.to_device(queue, np.arange(2000), allocator=mem_pool) + 4000
+
+    assert a_dev.allocator is mem_pool
+    assert b_dev.allocator is mem_pool
+
+    a_dev2 = cl_array.arange(queue2, 2000, dtype=np.float32, allocator=mem_pool)
+    b_dev2 = cl_array.to_device(queue2, np.arange(2000), allocator=mem_pool) + 4000
+
+    assert a_dev2.allocator is mem_pool
+    assert b_dev2.allocator is mem_pool
+
+    np.testing.assert_allclose((a_dev+b_dev).get(), (a_dev2+b_dev2).get())
```

Comment:
This is dangerous and wrong, and I suspect it is what caused the crashes on the past few test runs. It's wrong because there is work that is explicitly not synchronized with memory deallocation. I should have looked at this more carefully.

Reply:
I'll fix it, but we also need to figure out whether the blast radius of mistakes of this type can be limited.

Reply:
I've pushed a fix, but I haven't yet made a way to catch issues of this type generically. The main problem is getting access to a queue for comparison when setting kernel arguments. I'll bump this to an issue for now. #601